source/server/hot_restarting_child.h

```diff
@@ -14,6 +14,28 @@ class HotRestartingChild : HotRestartingBase {
   HotRestartingChild(int base_id, int restart_epoch, const std::string& socket_path,
                      mode_t socket_mode);
 
+  // A structure to record the set of registered UDP listeners keyed on their addresses,
+  // to support QUIC packet forwarding.
+  class UdpForwardingContext {
+  public:
+    using ForwardEntry = std::pair<const Network::Address::Instance*, Network::UdpListenerConfig*>;
+
+    // Returns the address and UdpListenerConfig associated with the given address.
+    // The addresses are not necessarily identical, as e.g. the listener might be listening on
+    // 0.0.0.0.
+    absl::optional<ForwardEntry>
```

Comment:
It's worth to comment about the threading model of `registerListener()` and `getListenerForDestination()`